Burak-last-hashtag-redirect

myniqx opened the pull request:
https://github.com/freedomcombination/monorepo/issues/1155

7alip commented on apps/trend-rights/middleware.ts:
```diff
@@ -6,18 +8,42 @@ const corsHeaders = {
-export function middleware(request: NextRequest) {
-  if (request.method === 'OPTIONS') {
-    return NextResponse.json({}, { headers: corsHeaders })
+const dontRedirectIfDevelopment = true // prevents redirecting if in development
```
Let's redirect in dev mode as well. We can easily check it always

7alip commented on apps/trend-rights/middleware.ts:
```diff
@@ -6,18 +8,42 @@ const corsHeaders = {
+  })
+
+  if (isCurrentDateWithinInterval) {
+    return NextResponse.redirect('/hashtags/' + slug)
```
https://nextjs.org/docs/pages/building-your-application/routing/redirecting#nextresponseredirect-in-middleware

In Next.js pages router example, they use redirect like this:

`return NextResponse.redirect(new URL('/login', request.url))`

7alip commented on apps/trend-rights/middleware.ts:
```diff
@@ -6,18 +8,42 @@ const corsHeaders = {
+    end: hashtagDatetimePlus12Hours,
+  })
+
+  if (isCurrentDateWithinInterval) {
```
Check if redirected already

```ts
const isRedirected = slug === cookieSlug
```

7alip reviewed (commented):
Thanks

myniqx commented:
@7alip im sorry i didnt get review request back. i rewrote almost everthing. its better.
what is working:
- reads from edge-config 
- if it tries to fetch the key more than once in 10 seconds, it skips (cookie named dofetch)
- if it already redirected (cookie named redirectedslug) or already in hashtags/:slug, it skips.
- in backend afterCreate and afterUpdate without check any condition last hashtag ,in teory, added to edge config.
what is not working:
-last hashtag cannot be updated to edge config.

i test it under there condition:
- i return static last-hashtag and do testing and get those result (above)
- and also backend works until set function. it returns an error 403. it seems that token in connection string is working while reading, but when create or update, it doesnt work.
if you paste the connection string you gave me to address bar you can easly see its working.
but when write (https://vercel.com/docs/storage/edge-config/vercel-api#update-your-edge-config-items)
this is the exact same error : 
```
{
  "error": {
    "code": "forbidden",
    "message": "The request is missing an authentication token",
    "missingToken": true
  }
}
```
maybe with the correct token, this ll work.

myniqx commented:
@7alip now its better :)
in middleware:
i added matcher, with a litlle bir more filter it works without do-fetch cookie. so i remove it.
vercel's get function is now working. it seems it was a matter releated with turbo. 
and i added another edge key for development mode, 'last-hashtag-dev'.

7alip reviewed (approved):
Thanks
